Content.Server/GameObjects/Components/Nutrition/CreamPiedComponent.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Content.Server.GameObjects.Components.Mobs;
+using Content.Server.Utility;
+using Content.Shared.Chemistry;
+using Content.Shared.GameObjects.Components.Nutrition;
+using Content.Shared.Interfaces;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Localization;
+using Robust.Shared.ViewVariables;
+using Serilog;
+
+namespace Content.Server.GameObjects.Components.Nutrition
+{
+    [RegisterComponent]
+    public class CreamPiedComponent : SharedCreamPiedComponent, IReagentReaction, IThrowCollide
+    {
+        private bool _creamPied;
+
+        [ViewVariables]
+        public bool CreamPied
+        {
+            get => _creamPied;
+            private set
+            {
+                _creamPied = value;
+                if (Owner.TryGetComponent(out AppearanceComponent appearance))
+                {
+                    appearance.SetData(CreamPiedVisuals.Creamed, CreamPied);
+                }
+            }
+        }
+
+        public void Wash()
+        {
+            if(CreamPied)
+                CreamPied = false;
+        }
+
+        public ReagentUnit ReagentReactTouch(ReagentPrototype reagent, ReagentUnit volume)
+        {
+            switch (reagent.ID)
+            {
+                case "chem.SpaceCleaner":
+                case "chem.H2O":
+                    Wash();
+                    break;
+            }
+
+            return ReagentUnit.Zero;
+        }
+
+        public void HitBy(ThrowCollideEventArgs eventArgs)
+        {
+            if (eventArgs.Target != Owner || !eventArgs.Thrown.TryGetComponent(out CreamPieComponent creamPie) || CreamPied) return;
```

Comment:
You can remove these target!=Owner, thrown!=Owner style checks now can't you?

Reply:
yeah, I can. I forgot to change this--